core/src/main/java/net/sf/hajdbc/dialect/h2/H2Dialect.java

```diff
@@ -151,4 +158,54 @@ protected boolean indicatesFailure(int code)
 	{
 		return failureCodes.contains(code);
 	}
+	
+	@Override
+	public DumpRestoreSupport getDumpRestoreSupport() {
+		return this;
+	}
+    
+	@Override
+	public <Z, D extends Database<Z>> void dump(D database, Decoder decoder, File file, boolean dataOnly) throws Exception {
+		try (Connection c = database.connect(decoder)) {
+			try (Statement s = c.createStatement()) {
+		      if (!s.execute("set EXCLUSIVE 1;")) {
+		    	  try (ResultSet rs = s.executeQuery("SCRIPT;")) {
+		    		  try (FileWriter fw = new FileWriter(file)) {
+			    		  while (rs.next()) {
+			    			 fw.write(rs.getString(1) + "\n");  
+			    		  }
+		    		  }
+		    	  } finally {
+		    		 s.execute("set EXCLUSIVE 0;");
+		    	  }
+		      }
+			} catch (Exception e) {
```

Comment:
Can you drop the catch block?  We don't want to write to System.err or System.out.
Let the caller handle the exception.